htdocs/modulebuilder/template/admin/myobject_extrafields.php

```diff
@@ -28,27 +29,25 @@
 
 // Load Dolibarr environment
 $res = 0;
+// Try main.inc.php into web root defined in DOCUMENT_ROOT, very frequently defined
+if (!$res && !empty($_SERVER["DOCUMENT_ROOT"])) {
+	$res = @include $_SERVER["DOCUMENT_ROOT"]."/main.inc.php";
+}
 // Try main.inc.php into web root known defined into CONTEXT_DOCUMENT_ROOT (not always defined)
 if (!$res && !empty($_SERVER["CONTEXT_DOCUMENT_ROOT"])) {
 	$res = @include $_SERVER["CONTEXT_DOCUMENT_ROOT"]."/main.inc.php";
 }
-// Try main.inc.php into web root detected using web root calculated from SCRIPT_FILENAME
-$tmp = empty($_SERVER['SCRIPT_FILENAME']) ? '' : $_SERVER['SCRIPT_FILENAME']; $tmp2 = realpath(__FILE__); $i = strlen($tmp) - 1; $j = strlen($tmp2) - 1;
-while ($i > 0 && $j > 0 && isset($tmp[$i]) && isset($tmp2[$j]) && $tmp[$i] == $tmp2[$j]) {
-	$i--; $j--;
-}
-if (!$res && $i > 0 && file_exists(substr($tmp, 0, ($i + 1))."/main.inc.php")) {
-	$res = @include substr($tmp, 0, ($i + 1))."/main.inc.php";
-}
-if (!$res && $i > 0 && file_exists(dirname(substr($tmp, 0, ($i + 1)))."/main.inc.php")) {
-	$res = @include dirname(substr($tmp, 0, ($i + 1)))."/main.inc.php";
-}
-// Try main.inc.php using relative path
-if (!$res && file_exists("../../main.inc.php")) {
-	$res = @include "../../main.inc.php";
-}
-if (!$res && file_exists("../../../main.inc.php")) {
-	$res = @include "../../../main.inc.php";
+// Try main.inc.php in all parent folders (cope with symlinks)
+if (!$res) {
+	$dolipath = dirname($_SERVER['SCRIPT_FILENAME']);
```

Comment:
Script_filename is not always defined, so we must have a fallback using the .. ../.. etc syntax.
Also script_filename and document_root are root if the server but does not take into account the case we are using an apache alias to access different web sites on the same virtual host but isolated with an alias.
There is a lot of cases to support:
- case web server is used throw a proxy
- case web server is using a cgi wrapper like fpm instead of a apache prefork wrapper
- case web server is used with a load repartition system layer
- case web server run with virtualhost on a different filesystem than the main apache biqnaries, when using symbolic links, when using hard links.
- case using some php security modules that hide some server variables
- case web server uses a special setup of some exotics panel managers like cpanel, plesk, ...
- case web server is using apache "directory alias" instead of virtual host to isolate servers...
And any combination of all of this !

Current handler to find the main.inc.php contains a lot of lines, yes, but is has 20 years of experience and has proved it works in all the situation (probably nearly hundred of different combination of situations).
We already experiment use of document_root and we know we must NOT use it.
Changing the main loader will surely break some cases, even if they are not frequent, when current is robust. It does not worth the gain.
Only the mutualization of the search using the .. then ../.. then ../../.. etc into a loop can be done at no risk.